angular-frontend/src/app/authorization/authorization.service.ts

```diff
@@ -0,0 +1,229 @@
+import {Injectable} from '@angular/core';
+import {HttpClient, HttpHeaders, HttpParams, HttpRequest} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {Router} from "@angular/router";
+import jwt_decode from 'jwt-decode';
+import {AccessTokenResponse, DecodedToken} from './AuthorizationTypes';
+import {randomBytes} from 'crypto';
+import * as cryptojs from "crypto";
+import base64url from 'base64url';
+import {Utils} from "../utils";
+import {AppConfigService} from "../app-config.service";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthorizationService {
+  private readonly authorizationEndpoint = '/oauth2/authorize';
+  private readonly codeChallengeMethod = 'S256';
+  private readonly responseType = 'code';
+  private readonly authorizationCode = 'authorization_code';
+  private readonly accessTokenRef = 'access_token';
```

Comment:
(Minor) Maybe it'd make sense to turn these `private readonly` fields into constants `const authorizationEndpoint = ...` outside the class but scoped the ts module. Maybe I'm too old school with this comment...

Reply:
I'm definitely open to this, but maybe not familiar enough with the pattern you're referring to. How do I scope it to the module?

Reply:
Those fields are constants as far as I understood. Correct? If they are not then the below obviously does not apply...

Move these out of the class and have them as:
```
const CODE_CHALLENGE_METHOD = 'S256';
const RESPONSE_TYPE = 'code';
const AUTHORIZATION_CODE = 'authorization_code';
const ACCESS_TOKEN_REF = 'access_token';

export class AppConfigService {
...
}
```

Reply:
Yes, they are constants. Oh, I see. Literally just move the fields out of the class. I can do that, but I'm not sure I see the benefit?

Reply:
If it is some sort of a sample app for people to use as starting point then it might make sense in terms of just code style... As I mentioned it is a minor thing and it is up to you'd like to make the change. 

If I recall correctly a private field on the class can still be changed because at the end it is just JS. Cast it to any and change the field. The scope constant though as a JS scope variable won't be accessible unless exported. I think.

If you like to leave this chunk as is - I don't mind :-)

Reply:
Cool, thank you for the explanation! I can change it :)